sound/soc/sof/ipc4-pcm.c

```diff
@@ -410,15 +411,31 @@ static void sof_ipc4_pcm_free(struct snd_sof_dev *sdev, struct snd_sof_pcm *spcm
 		pipeline_list = &spcm->stream[stream].pipeline_list;
 		kfree(pipeline_list->pipelines);
 		pipeline_list->pipelines = NULL;
+		kfree(spcm->stream[stream].private);
+		spcm->stream[stream].private = NULL;
 	}
 }
 
 static int sof_ipc4_pcm_setup(struct snd_sof_dev *sdev, struct snd_sof_pcm *spcm)
 {
 	struct snd_sof_pcm_stream_pipeline_list *pipeline_list;
 	struct sof_ipc4_fw_data *ipc4_data = sdev->private;
+	struct sof_ipc4_timestamp_info *stream_info;
+	bool support_info = true;
+	u32 abi_version;
+	u32 abi_offset;
 	int stream;
 
+	abi_offset = offsetof(struct sof_ipc4_fw_registers, abi_ver);
+	sof_mailbox_read(sdev, sdev->fw_info_box.offset + abi_offset, &abi_version,
+			 sizeof(abi_version));
+
+	if (abi_version < SOF_IPC4_FW_REGS_ABI_VER) {
+		dev_info(sdev->dev, "unsupported abi version %d for delay feature\n",
+			 abi_version);
+		support_info = false;
```

Comment:
Is this going to be printed for all not MTL platforms? Will non MTL platforms going to have this information, but still not supporting the timestamp?
Should this be debug, or even dropped?

Reply:
OK, so latest ADL, RPL (presumably TGL) supports this, so the `spcm->stream[stream].private` is going to be not NULL for these platforms as well.
But only MTL provides the `get_stream_position` callback, which means that only MTL will work.
Should we also toss in an additional ` && (sof_ops(sdev) && sof_ops(sdev)->get_stream_position)` check to block the functionality on platforms where it is not supported?